app/lib/BundlableLabelableBaseModelWithAttributes.php

```diff
@@ -722,8 +722,14 @@ public function set($pa_fields, $pm_value="", $pa_options=null) {
 		}
 		
 		if ($this->opo_idno_plugin_instance) {
+			if ($this->isChild()) {
+				$parent_idno = self::getIdnoForID($this->get($parent_id_fld));
+			}
+			if (($vs_parent_id_fld = $this->getProperty('HIERARCHY_PARENT_ID_FLD')) && isset($pa_fields[$vs_parent_id_fld]) && ($pa_fields[$vs_parent_id_fld] > 0)) {
+				$parent_idno = self::getIdnoForID($pa_fields[$vs_parent_id_fld]);
+			}
 			// If attempting to set parent_id, then flag record as child for id numbering purposes
-			$this->opo_idno_plugin_instance->isChild(((($vs_parent_id_fld = $this->getProperty('HIERARCHY_PARENT_ID_FLD')) && isset($pa_fields[$vs_parent_id_fld]) && ($pa_fields[$vs_parent_id_fld] > 0)) || ($this->isChild())) ? true : null);
+			$this->opo_idno_plugin_instance->isChild(isset($parent_idno) ? true : null, isset($parent_idno) ? $parent_idno : null);
```

Comment:
I was trying to mirror what was already there, but I think we should just not call isChild if we don't have $parent_idno